src/main/java/com/google/devtools/build/lib/remote/RemoteExecutionService.java

```diff
@@ -522,6 +534,31 @@ private ToolSignature getToolSignature(Spawn spawn, SpawnExecutionContext contex
         : null;
   }
 
+  private boolean shouldUseOutputPaths(Spawn spawn) {
+    var isRemoteExec = mayBeExecutedRemotely(spawn);
+    var isRemoteCached = mayBeCachedRemotely(spawn);
```

Comment:
This doesn't work for combined cache. And what happens if remote cache and remote executor return different API version?

I don't get why we need two server capabilities. It seem likes it only makes thing more complicated. Bazel should use the same API version for both remote cache and remote execution (if the two endpoints are different).

Reply:
> I don't get why we need two server capabilities. It seem likes it only makes thing more complicated. Bazel should use the same API version for both remote cache and remote execution (if the two endpoints are different).

So in the case of remote cache and remote executor differ, we know that we could have 2 server capabilities regardless.

When that happens, we _SHOULD_ run some validation against the 2 server capabilities to figure out what is the API version range. And in case the version ranges of Remote Cache and Remote Executor do not overlap, then we err out.

With that in mind, I will attempt to implement that logic in RemoteModule. Then in RemoteExecutionService, we should be able to use one server capabilities instead of 2.

Reply:
SGTM!